crates/ruff/src/rules/flake8_todo/rules.rs

```diff
@@ -0,0 +1,170 @@
+use std::iter::Peekable;
+
+use once_cell::sync::Lazy;
+
+use regex::{CaptureMatches, Regex};
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::types::Range;
+use rustpython_parser::ast::Location;
+use rustpython_parser::lexer::LexResult;
+use rustpython_parser::Tok;
+
+#[violation]
+pub struct InvalidTODOTag {
```

Comment:
The [Rust naming convention](https://rust-lang.github.io/api-guidelines/naming.html) suggests treating acronyms as one word, so all these should be e.g. `InvalidTodoTag`.

Reply:
Done